docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,422 @@
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to setup workload identity")
+			}
+
+			client := subscriptions.NewClient()
+
+			// setCredentialsForWorkloadIdentity just setups the 
+			// PublicCloud env URLs 
+			params.AzureClients.setCredentialsForWorkloadIdentity(ctx, params.AzureCluster.Spec.SubscriptionID, params.AzureCluster.Spec.AzureEnvironment)
+			client.Authorizer = azidext.NewTokenCredentialAdapter(cred, []string{"https://management.azure.com//.default"})
+			params.AzureClients.Authorizer = client.Authorizer
+
+```
+
+**NOTE:**
+`azidext.NewTokenCredentialAdapter` is used to get a authoriser in order to add to the existing code workflow to adapt an azcore.TokenCredential type to an autorest.Authorizer type.
+
+Also a go file e.g `workload_identity.go` in the `identity` package dealing with AZWI functionality.
+
+```go
+
+type workloadIdentityCredential struct {
+	assertion string
+	file      string
+	cred      *azidentity.ClientAssertionCredential
+	lastRead  time.Time
+}
+
+type workloadIdentityCredentialOptions struct {
+	azcore.ClientOptions
+}
+
+func newWorkloadIdentityCredential(tenantID, clientID, file string, options *workloadIdentityCredentialOptions) (*workloadIdentityCredential, error) {
+	w := &workloadIdentityCredential{file: file}
+	cred, err := azidentity.NewClientAssertionCredential(tenantID, clientID, w.getAssertion, &azidentity.ClientAssertionCredentialOptions{ClientOptions: options.ClientOptions})
+	if err != nil {
+		return nil, err
+	}
+	w.cred = cred
+	return w, nil
+}
+
+func (w *workloadIdentityCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
+	return w.cred.GetToken(ctx, opts)
+}
+
+func (w *workloadIdentityCredential) getAssertion(context.Context) (string, error) {
+	if now := time.Now(); w.lastRead.Add(5 * time.Minute).Before(now) {
+		content, err := os.ReadFile(w.file)
+		if err != nil {
+			return "", err
+		}
+		w.assertion = string(content)
+		w.lastRead = now
+	}
+	return w.assertion, nil
+}
+
+```
+
+### <a name='OpenQuestions'></a>Open Questions
+
+#### <a name='Howtomultitenancy'></a>1. How to achieve multi-tenancy?
+
+The identity is tied to the client ID which can be supplied via AzureClusterIdentity. In case identity is not found in AzureClusterIdentity client ID from the service account annotation
+`azure.workload.identity/client-id` will be used.
+
+#### <a name='Howtodistributekeys'></a>2. How to distribute key pair to management cluster?
+
+Keys can be distributed by using the CABPK feature by creating a secret with name `<cluster-name>-sa` encompassing the key pair. More details on it [here](https://cluster-api.sigs.k8s.io/tasks/certs/using-custom-certificates.html)
+
+#### <a name='UserExperience'></a>3. User Experience
+
+Though AZWI has a lot of advantages as compared to AZWI, setting up AZWI involves couple of manual step and it can impact the user experience.
```

Comment:
```suggestion
Though AZWI has a lot of advantages as compared to AAD-POD-IDENTITY, setting up AZWI involves couple of manual step and it can impact the user experience.
```
Did you mean to say the above?